numpy/polynomial/polynomial.py

```diff
@@ -407,7 +408,10 @@ def polydiv(c1, c2):
     if lc1 < lc2:
         return c1[:1]*0, c1
     elif lc2 == 1:
-        return c1/c2[-1], c1[:1]*0
+        try:            
+            return c1/c2[-1], c1[:1]*0
+        except RuntimeWarning:
```

Comment:
If this is where the warning is produced, perhaps the context manager for warning suppression could be placed nearby instead of doing this?

Reply:
Yep, good idea. I've refactored this code block to be more localized in terms of warning suppression now:
```
sup = suppress_warnings()
sup.filter(RuntimeWarning)
    with sup:
        return c1/c2[-1], c1[:1]*0 
```